### DISCUSSION TITLE
cursorless - knausj_talon compatibility

### DISCUSSION THREAD

**Discussion by codecat555** [cursorless-dev/cursorless#314] [General]
Loving cursorless!

I just ran into a knausj_talon compatibility issue with cursorless, and @pokey invited me to share it here. I'd be interested in hearing about any other tweaks people have made to avoid conflicts.

Cursorless placed an uncolored marker above the number 8 in my text and I wanted to clear that word, so I said 'clear 8'. Then, vscode jumped to line 8 (not the target line) and erased it - not what I expected.

I found this command and just changed it so the '[line]' part is not optional - https://github.com/knausj85/knausj_talon/blob/65ccd94b8fe97056a34b858b2f1f261cc5269c3f/text/line_commands.talon#L16 .

**Comment by pokey** [comment]
@codecat555 fwiw we fixed the specific issue you raised by changing the default term for "clear" to "change", but that's certainly not the only knausj conflict, so let's leave this discussion open.  The other ones I know of are:

- Uses of "that", eg "copy that", "paste that", "phones that", etc
- Uses of "line", eg "copy line", "clear line", etc, although these commands have roughly the same behaviour in knausj
- "phones made", "phones whale", etc, where our "phones \<target\>" conflicts with knausj's "phones \<homophone\>"

### Possible solutions

- Change commands in cursorless
- Change commands in knausj
- Convert these commands to use an action, and redefine the action from cursorless

cc/ @AndreasArvidsson @knausj85 have I missed any conflicts?

**Reply by AndreasArvidsson** [reply]
That's all to my knowledge, but I'm not an expert on knausj commands

**Comment by alexhwoods** [comment]
Note - `lang/tags/comment_line.talon` defines comment commands that conflict with `comment`, particularly `comment <user.text>`

**Reply by alexhwoods** [reply]
Similarly, a lot of `knausj_talon/text/line_commands.talon` can be commented out for Cursorless users.
- `cut [line] <number>` is particularly conflicting
